packages/wonder-blocks-core/components/no-ssr.test.js

```diff
@@ -11,34 +11,60 @@ describe("NoSSR", () => {
         unmountAll();
     });
 
-    test("renders a placeholder first, then the actual content", (done) => {
+    test("renders a placeholder first, then the actual content", async () => {
         // Arrange
         const mockPlaceholder = jest.fn(() => null);
-
-        const arrangeAct = (assert) => {
+        await new Promise((resolve) => {
             const nodes = (
                 <NoSSR placeholder={mockPlaceholder}>
                     {() => {
-                        assert();
+                        resolve();
                         return null;
                     }}
                 </NoSSR>
             );
 
             // Act
             mount(nodes);
-        };
+        });
 
-        const assert = () => {
-            // Assert
-            // This was called from within the NoSSR children render prop.
-            // Therefore, if the placeholder has been called, it must have
-            // been called first.
-            expect(mockPlaceholder).toHaveBeenCalledTimes(1);
-            done();
-        };
+        // Assert
+        // Our promise doesn't resolve until the children render, therefore
+        // we don't get here until that and so if the placeholder has been
+        // called, it must have been called first.
+        expect(mockPlaceholder).toHaveBeenCalledTimes(1);
```

Comment:
Given the title of the test I would expect to see and `expect` for what was rendered into `nodes`.

Reply:
Well, this isn't testing that the render is working, just that the call to render is being called. Snapshot tests etc. cover the actual return result getting used in the output. This is more about ensuring order of events match our expectations.

Reply:
Renamed the test